src/vehicle_gui.cpp

```diff
@@ -3192,7 +3214,22 @@ struct VehicleViewWindow : Window {
 	{
 		if (str == nullptr) return;
 
-		Command<CMD_RENAME_VEHICLE>::Post(STR_ERROR_CAN_T_RENAME_TRAIN + Vehicle::Get(this->window_number)->type, this->window_number, str);
+		const Vehicle *v = Vehicle::Get(this->window_number);
+		switch (query) {
+			case QueryType::RenameVehicle:
+				Command<CMD_RENAME_VEHICLE>::Post(STR_ERROR_CAN_T_RENAME_TRAIN + v->type, v->index, str);
+				break;
+			case QueryType::RenameGroup:
+				if (v->group_id == DEFAULT_GROUP)
+					Command<CMD_ADD_VEHICLE_GROUP>::Post(STR_ERROR_GROUP_CAN_T_ADD_VEHICLE, NEW_GROUP, v->index, true /* add shared */);
+				if (v->group_id != DEFAULT_GROUP) // Not guaranteed if CMD_ADD_VEHICLE_GROUP fails.
+					Command<CMD_ALTER_GROUP>::Post(STR_ERROR_GROUP_CAN_T_RENAME, AlterGroupMode::Rename, v->group_id, 0, str);
+				break;
+			default:
+				; /* Not supposed to happen. */
+		}
+
+		query = QueryType::None;
```

Comment:
```suggestion
		this->query = QueryType::None;
```
but as mentioned in another comment `QueryType::None` is most likely unneeded.